src/main/java/com/telefonica/iot/cygnus/sinks/OrionKafkaSink.java

```diff
@@ -128,16 +118,13 @@ public void start() {
             props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
             props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
             props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-            persistenceBackend = new KafkaProducer<String, String>(props);
+            persistenceBackend = new KafkaBackendImpl(props);
             LOGGER.debug("[" + this.getName() + "] Kafka persistence backend (KafkaProducer) created");
         } catch (Exception e) {
             LOGGER.error("Error while creating the Kafka persistence backend (KafkaProducer). Details="
                     + e.getMessage());
         } // try catch
 
-        // creat the name API
-        topicAPI = new TopicAPI();
-
         // create the Zookeeper client
         zookeeperClient = new ZkClient(zookeeperEndpoint, 10000, 10000, ZKStringSerializer$.MODULE$);
```

Comment:
I think the best is moving the creation of the Zookeeper client to KafkaBackendiImpl. This way the KavkaBackendImpl methods invocation is as simple as pasing the topic name when asking if it exists, etc.